src/eddymotion/utils.py

```diff
@@ -0,0 +1,149 @@
+# emacs: -*- mode: python; py-indent-offset: 4; indent-tabs-mode: nil -*-
+# vi: set ft=python sts=4 ts=4 sw=4 et:
+#
+# Copyright 2022 The NiPreps Developers <nipreps@gmail.com>
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# We support and encourage derived works from this project, please read
+# about our expectations at
+#
+#     https://www.nipreps.org/community/licensing/
+#
+"""Utils to sort the DWI data volume indices"""
+
+import numpy as np
+
+
+def linear_action(size=None, **kwargs):
+    """
+    Sort the DWI data volume indices linearly
+
+    Parameters
+    ----------
+    size : :obj:`int`
+        Number of volumes in the dataset
+        (for instance, the number of orientations in a DWI)
+
+    Returns
+    -------
+    :obj:`list` of :obj:`int`
+        The sorted index order.
+
+    Examples
+    --------
+    >>> list(linear_action(10))
+    [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
+
+    """
+    if size is None and 'bvals' in kwargs:
+        size = len(kwargs['bvals'])
+    if size is None:
+        raise TypeError("Cannot build iterator without size")
+
+    return range(size)
+
+
+def random_action(size=None, **kwargs):
+    """Sort the DWI data volume indices.
+
+    Parameters
+    ----------
+    size : :obj:`int`
+        Number of volumes in the dataset
+        (for instance, the number of orientations in a DWI)
+
+    Returns
+    -------
+    :obj:`list` of :obj:`int`
+        The sorted index order.
+    """
+
+    if size is None and 'bvals' in kwargs:
+        size = len(kwargs['bvals'])
+    if size is None:
+        raise TypeError("Cannot build iterator without size")
+
+    _seed = kwargs.get('seed', None)
+    if kwargs.get('seed', None) or kwargs.get('seed', None) == 0:
+        _seed = 20210324 if kwargs.get('seed', None) is True else kwargs.get('seed', None)
```

Comment:
I don't get this (??)

Reply:
It's inspired from the _sort_dwdata_indices() function in estimator.py. I thought that in the future, we would want to externalize the initialization of the seed in the random generator instead of the backend function _sort_dwdata_indices() but feel free to remove if it's not the right place for this.

However, there seems to be a bug in the original function -- when the seed argument is set to 0, the condition "if seed or seed == 0:" considers zero as False, and the value of the seed is not set correctly to 20210324, but instead stays at 0. I am not sure if this bug is only present in some Python versions, but I am currently using version 3.11. Should I post an issue about this, or is it okay if I propose a fix this in this PR?

Reply:
I think you are referring to this code added by @jhlegarreta:

https://github.com/nipreps/eddymotion/blob/a1bcacd6117a9f9ce95424dae7fb17c893a885a1/src/eddymotion/estimator.py#L261-L263

Here, I think the purpose is to set the default value (20210324) if (and only if) the seed argument is a `True` value, and when seed is the integer zero (`0`), then set the seed to zero (as it is an acceptable value).

Arguably, this code probably would have worked just fine if you do:

```
seed = 20210324 if seed is True else seed
```

That would not be fully equivalent, as `False` would remain `False` with my code. However, `False` is actually an acceptable value by the random number generator seed. I will send a suggestion with this (since this has become outdated, I cannot do it here).

Reply:
False is cast to zero -- I'm going to update my suggestion so that False is made None.

Reply:
Ok, thanks for clarifying. I wondered if zero was avoided for another reason.